### PR TITLE
[KOGITO-816] - Operator provisioning for Task Console

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submiting your Pull Request :heart:! 

Jira issue : https://issues.redhat.com/browse/KOGITO-816, https://issues.redhat.com/browse/KOGITO-1006

With this implementation user could easily deploy Task-console using Kogito-operator through below ways :
1. Using Openshift UI
2. Create task-console deployment using below yaml:
```
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoSupportingService
metadata:
  name: kogito-task-console
spec:
  serviceType: "TaskConsole"
  replicas: 1
  image: "quay.io/vajain/kogito-task-console:1.0"
```
3. Create using Kogito-cli
```
kogito install task-console --image quay.io/vajain/kogito-task-console:1.0
```
Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/controller/kogitosupportingservice/kogitodataindex_controller.go [hunk 1]
```diff
@@ -174,7 +174,10 @@ func (r *ReconcileKogitoDataIndex) Reconcile(request reconcile.Request) (result
 	if resultErr = infrastructure.InjectDataIndexURLIntoKogitoRuntimeServices(r.client, instance.Namespace); resultErr != nil {
 		return
 	}
-	if resultErr = infrastructure.InjectDataIndexURLIntoMgmtConsole(r.client, instance.Namespace); resultErr != nil {
+	if resultErr = infrastructure.InjectDataIndexURLIntoSupportingService(r.client, instance.Namespace, appv1alpha1.MgmtConsole); resultErr != nil {
+		return
+	}
+	if resultErr = infrastructure.InjectDataIndexURLIntoSupportingService(r.client, instance.Namespace, appv1alpha1.TaskConsole); resultErr != nil {
```
Comment (ricardozanini) [inline pkg/controller/kogitosupportingservice/kogitodataindex_controller.go:180]: Maybe we could refactor this function to receive a slice of resources instead?

```go
infrastructure.InjectDataIndexURLIntoSupportingService(r.client, instance.Namespace, appv1alpha1.TaskConsole, appv1alpha1.MgmtConsole)
```

File: pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go [hunk 2]
```diff
@@ -229,4 +229,5 @@ var kogitoSupportingServices = map[v1alpha1.ServiceType]SupportingServiceResourc
 	v1alpha1.Explainablity: &ExplainabilitySupportingServiceResource{},
 	v1alpha1.TrustyAI:      &TrustyAISupportingServiceResource{},
 	v1alpha1.TrustyUI:      &TrustyUISupportingServiceResource{},
+	v1alpha1.TaskConsole:   &TaskConsoleSupportingServiceResource{},
```
Comment (radtriste) [inline pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go:232]: I think we should put them in alphabetical order, will be easier to find them :)

**Review by ricardozanini — COMMENTED** [review]
LGTM, just one small nitpick.

**Comment by ricardozanini** [comment]
@vaibhavjainwiz one more thing: please update the RELEASE_NOTES.md file.

**Review by radtriste — COMMENTED** [review]
looks good. one comment about code oraganization.

Gonna create a BDD test for that to test it
In the mean time can you add an example yaml file for task console ?

File: pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go (hunk 2 above)
Comment (radtriste) [reply pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go:232]: I guess the alphabetical order would go also for other files where the different supporting services are configured/listed

**Review by radtriste — APPROVED** [review]
BDD tests passed

File: pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go (hunk 2 above)
Comment (vaibhavjainwiz) [reply pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go:232]: done

File: pkg/controller/kogitosupportingservice/kogitodataindex_controller.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply pkg/controller/kogitosupportingservice/kogitodataindex_controller.go:180]: done

**Comment by vaibhavjainwiz** [comment]
> @vaibhavjainwiz one more thing: please update the RELEASE_NOTES.md file.

done

**Comment by vaibhavjainwiz** [comment]
> looks good. one comment about code oraganization.
> 
> Gonna create a BDD test for that to test it
> In the mean time can you add an example yaml file for task console ?

added yaml file under examples

**Comment by radtriste** [comment]
similar to https://github.com/kiegroup/kogito-cloud-operator/pull/633#issuecomment-718487735

/jenkins test

**Comment by radtriste** [comment]
@ricardozanini @spolti can you have a look ?
Once this one merged I will set the one on BDD tests as ready for review

**Review by ricardozanini — APPROVED** [review]

**Review by Kaitou786 — APPROVED** [review]

**Review by spolti — APPROVED** [review]

File: pkg/infrastructure/kogitosupporting_service.go [hunk 3]
```diff
@@ -44,3 +45,29 @@ func getKogitoSupportingService(client *client.Client, namespace string, resourc
 	}
 	return nil, nil
 }
+
+// getKogitoRuntimeDeployments gets all dcs owned by KogitoRuntime services within the given namespace
```
Comment (sutaakar) [inline pkg/infrastructure/kogitosupporting_service.go:49]:
```suggestion
// getSupportingServiceDeployment gets deployment owned by supporting service within the given namespace
```

**Review by sutaakar — APPROVED** [review]

**Comment by ricardozanini** [comment]
@vaibhavjainwiz could you please rebase so we can merge! \o/

**Comment by vaibhavjainwiz** [comment]
> @vaibhavjainwiz could you please rebase so we can merge! \o/

done

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**